pkg/runtime/meta/meta.go

```diff
@@ -96,3 +108,28 @@ func ContainsNamespace(items []common.NameValuePair) bool {
 	}
 	return false
 }
+
+// AddWasmStrictSandbox adds global wasm strict sandbox configuration to component metadata.
+// When strict sandbox is enabled, WASM components always run in strict mode regardless of their configuration.
+// When strict sandbox is disabled or unset, keep the original component configuration.
+func (m *Meta) AddWasmStrictSandbox(comp *compapi.Component) {
+	// If the global strict sandbox is disabled(or unset), do nothing.
+	if m.strictSandbox == false {
+		return
+	}
+
+	// If the metadata already contains the strict sandbox key, update the value to global strict sandbox config.
+	for i, c := range comp.Spec.Metadata {
+		if strings.EqualFold(c.Name, WasmStrictSandboxMetadataKey) {
+			comp.Spec.Metadata[i].SetValue([]byte(strconv.FormatBool(m.strictSandbox)))
```

Comment:
If we're here, the value can only be "true"

```suggestion
			comp.Spec.Metadata[i].SetValue([]byte("true"))
```

Reply:
Done.